pkg/selectors/kernel.go

```diff
@@ -632,30 +632,30 @@ func ParseMatchCapabilityChanges(k *KernelSelectorState, actions []v1alpha1.Capa
 	return nil
 }
 
-func ParseMatchBinary(k *KernelSelectorState, b *v1alpha1.BinarySelector) error {
+func ParseMatchBinary(k *KernelSelectorState, b *v1alpha1.BinarySelector, selIdx int) error {
 	op, err := SelectorOp(b.Operator)
 	if err != nil {
 		return fmt.Errorf("matchBinary error: %w", err)
 	}
 	if op != SelectorOpIn && op != SelectorOpNotIn {
 		return fmt.Errorf("matchBinary error: Only In and NotIn operators are supported")
 	}
-	k.SetBinaryOp(op)
+	k.SetBinaryOp(selIdx, op)
 	for _, s := range b.Values {
 		if len(s) > 255 {
```

Comment:
do we still need this check? I still need to check the rest of the change, but the binary name does not go to kernel now, right?

Reply:
You mean the `len(s) > 255` right? They go into `names_map` and this length is due to the map key size (struct names_map_key) as shown in https://github.com/cilium/tetragon/blob/e7449cf7d1e5dfd8070b932cfa3abbc3b49ee3d8/bpf/lib/generic.h#L67-L76